apps/src/templates/progress/ProgressLessonTeacherInfo.story.jsx

```diff
@@ -114,178 +154,103 @@ const style = {
-              <div style={style}>
-                <ProgressLessonTeacherInfo
-                  lesson={lessons(state.progress)[2]}
-                />
-              </div>
-            </Provider>
-          );
-        }
-      },
-      {
-        name:
-          'hideable allowed, nonlockable lesson with lesson plan, without lesson url',
-        story: () => {
-          const store = createStore();
-          const state = store.getState();
-          return (
-            <Provider store={store}>
-              <div style={style}>
-                <ProgressLessonTeacherInfo
-                  lesson={lessons(state.progress)[1]}
-                />
-              </div>
-            </Provider>
-          );
-        }
-      },
-      {
-        name:
-          'hideable allowed, nonlockable lesson with lesson plan, with lesson url',
-        story: () => {
-          const store = createStore();
-          const state = store.getState();
-          return (
-            <Provider store={store}>
-              <div style={style}>
-                <ProgressLessonTeacherInfo
-                  lesson={lessons(state.progress)[1]}
-                />
-              </div>
-            </Provider>
-          );
-        }
-      },
-      {
-        name:
-          'hideable not allowed, nonlockable lesson with no lesson plan, without lesson url',
-        description: "shouldn't render anything",
-        story: () => {
-          const store = createStore({allowHidden: false});
-          const state = store.getState();
-          return (
-            <Provider store={store}>
-              <div style={style}>
-                <ProgressLessonTeacherInfo
-                  lesson={lessons(state.progress)[3]}
-                />
-              </div>
-            </Provider>
-          );
-        }
-      },
-      {
-        name:
-          'hideable not allowed, nonlockable lesson with no lesson plan, with lesson url',
-        story: () => {
-          const store = createStore({allowHidden: false});
-          const state = store.getState();
-          return (
-            <Provider store={store}>
-              <div style={style}>
-                <ProgressLessonTeacherInfo
-                  lesson={lessons(state.progress)[3]}
-                />
-              </div>
-            </Provider>
-          );
-        }
-      }
-    ]);
+export default {
+  title: 'ProgressLessonTeacherInfo',
+  component: ProgressLessonTeacherInfo
 };
+
+const store = createStore();
+const loadingStore = createStore({preload: true});
+const nonVerifiedTeacherStore = createStore({teacherVerified: false});
+const hiddenStore = createStore({allowHidden: false});
+
+const state = store.getState();
+const loadingState = loadingStore.getState();
+const nonVerifiedStoreState = nonVerifiedTeacherStore.getState();
+const hiddenStoreState = hiddenStore.getState();
+
+export const Loading = () => (
+  <Provider store={loadingStore}>
+    <div style={style}>
```

Comment:
Is it worth breaking this out into a template? I know not all of the stories use the same store but a lot do

Reply:
Did this! I made a more abstracted template so I could just feed in the store, state, and lessonIndex so that they could all use it.